### PR TITLE
Add Runnable.get_graph() to get a graph representation of a Runnable

### PR DESCRIPTION
**Pull request by nfcampos**
It can be drawn in ascii with Runnable.get_graph().draw()

### CODE REVIEW

File: libs/core/langchain_core/runnables/graph.py
Comment (baskaryan) [inline libs/core/langchain_core/runnables/graph.py:1]: if this file and class aren't meant to be used directly should we rename _graph.py and _Graph?

Comment (baskaryan) [reply libs/core/langchain_core/runnables/graph.py:1]: prevents them from getting documented as well

Comment (baskaryan) [reply libs/core/langchain_core/runnables/graph.py:1]: same with other file

File: libs/core/langchain_core/runnables/graph.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        self.edges.append(edge)
+        return edge
+
+    def extend(self, graph: Graph) -> None:
+        """Add all nodes and edges from another graph.
+        Note this doesn't check for duplicates, nor does it connect the graphs."""
+        self.nodes.update(graph.nodes)
+        self.edges.extend(graph.edges)
+
+    def first_node(self) -> Optional[Node]:
```
Comment (baskaryan) [inline libs/core/langchain_core/runnables/graph.py:61]: nit: source_node and sink_node are more intuitive to me than first_node and last_node but maybe just me

**Review by baskaryan — APPROVED** [review]

File: libs/core/langchain_core/runnables/graph.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+from dataclasses import dataclass, field
+from typing import Dict, List, NamedTuple, Optional, Type, Union
+from uuid import uuid4
+
+from langchain_core.pydantic_v1 import BaseModel
+from langchain_core.runnables.base import Runnable
+from langchain_core.runnables.graph_draw import draw
+
+
+class Edge(NamedTuple):
```
Comment (hinthornw) [inline libs/core/langchain_core/runnables/graph.py:12]: Is there a benefit to using named tuple here instead of other data classes?

Comment (eyurtsev) [reply libs/core/langchain_core/runnables/graph.py:12]: i also prefer data classes over named tuples with frozen=True

**Comment by eyurtsev** [comment]
@nfcampos is there a way to inspect .transform() implementation and identify / inform the user whether a given chain will be able to stream()? This might also aid in debugging issues for users since folks don't always correctly identify if something is streaming or not

File: libs/core/langchain_core/runnables/graph.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        last_node = self.last_node()
+        if last_node:
+            if (
+                len(self.nodes) == 1
+                or len([edge for edge in self.edges if edge.target == last_node.id])
+                == 1
+            ):
+                self.remove_node(last_node)
+
+    def draw_ascii(self) -> str:
```
Comment (akira) [inline libs/core/langchain_core/runnables/graph.py:111]: should these be separate renderers and support different rendering types?  for example ascii, mermaid, graphviz, etc.

Comment (nfcampos) [reply libs/core/langchain_core/runnables/graph.py:111]: yea that's why its a separate method

**Pull request merged**